apps/estuary/lib/estuary_web/live/streaming_event_live_view.ex

```diff
@@ -7,23 +7,32 @@ defmodule EstuaryWeb.StreamingEventLiveView do
   def render(assigns) do
     ~L"""
     <div class="events-index">
-      <h1 class="events-index__title">All Events</h1>
+      <h1 class="events-index__title">All Streaming Events</h1>
       <%= live_component(@socket, Table, id: :events_table, events: @events, order: @order) %>
     </div>
     """
   end
 
   def mount(_params, _session, socket) do
     EstuaryWeb.Endpoint.subscribe(@updated_event_stream)
-    {:ok, assign(socket, events: [], order: {"create_ts", "desc"}, params: %{})}
+    {:ok, assign(socket, events: nil, order: {"create_ts", "desc"}, params: %{})}
   end
 
   def handle_info(%{topic: @updated_event_stream, payload: %{events: events}}, socket) do
     updated_events =
-      (events ++ socket.assigns.events)
+      validate_events(events, socket.assigns.events)
```

Comment:
Using [List.wrap/1](https://hexdocs.pm/elixir/List.html#wrap/1) on socket events could help simplify this logic.